cogs/subjects.py

```
@@ -0,0 +1,272 @@
+                                                  self.db)
+
+    @subjects.command(pass_context=True,
+                      name="category")
+    async def subjects_category(self, ctx: Context, category: int):
+        """
+                Saves the group separator role:
+
+                Args:
+                    ctx: The command context provided by the discord.py wrapper.
+
+                    category: The category id for the channels.
+                """
+
+        db = PrimitiveMongoData(CollectionEnum.CATEGORIES)
+        key = ConfigurationNameEnum.SUBJECTS_CATEGORY
+        msg = "category"
+        await StudySubjectUtil.update_category_and_separator(category, ctx, db, key, msg)
+
+    @subjects.command(pass_context=True,
+                      name="separator")
+    async def subjects_separator(self, ctx: Context, role):  # parameter only for pretty help.
+        """
+        Saves the group separator role:
+
+        Args:
+            ctx: The command context provided by the discord.py wrapper.
+
+            role: The mentioned role.
+        """
+        if len(ctx.message.role_mentions) != 1:
+            raise BadArgument
+
+        role: Role = ctx.message.role_mentions[0]
+        key = ConfigurationNameEnum.SUBJECTS_SEPARATOR_ROLE
+
+        db = PrimitiveMongoData(CollectionEnum.ROLES)
+        msg = "separator"
+        await StudySubjectUtil.update_category_and_separator(role.id, ctx, db, key, msg)
+
+    async def get_possible_subjects(self, ctx: Context,
+                                    member: Union[Member, User],
+                                    roles: list[Role]) -> list[Role]:
+        study_group: list[Role] = [document.role for document in
+                                   await SubjectsOrGroups(self.bot, SubjectsOrGroupsEnum.GROUP).find({}) if
+                                   document.role in roles]
+        if not study_group:
+            embed = Embed(title="No Study group assigned",
```

This should throw an error so that it can be treated as an error.